modules/yieldmoBidAdapter.js

```diff
@@ -175,169 +229,223 @@ function getPageDescription() {
     }
-    return false;
-  } catch (e) {
-    return false;
+  };
+
+  const videoParams = utils.deepAccess(bidRequest, 'params.video');
+  Object.keys(videoParams)
+    .filter(param => includes(OPENRTB_VIDEO_BIDPARAMS, param))
+    .forEach(param => imp.video[param] = videoParams[param]);
+
+  if (videoParams.skippable) {
+    imp.video.skip = 1;
   }
+
+  return imp;
 }
 
 /**
- * @returns true if there is an AMP context object
+ * @param {BidRequest} bidRequest bidder request object.
+ * @return [number, number] || null Player's width and height, or undefined otherwise.
  */
-function isAmp() {
-  try {
-    const ampContext = window.context || window.parent.context;
-    if (ampContext && ampContext.pageViewId) {
-      return ampContext;
-    }
-    return false;
-  } catch (e) {
-    return false;
+function extractPlayerSize(bidRequest) {
+  const sizeArr = utils.deepAccess(bidRequest, 'mediaTypes.video.playerSize');
+  if (utils.isArrayOfNums(sizeArr, 2)) {
+    return sizeArr;
+  } else if (utils.isArray(sizeArr) && utils.isArrayOfNums(sizeArr[0], 2)) {
+    return sizeArr[0];
   }
+  return null;
 }
 
 /**
- * @returns true if the environment is a SafeFrame.
+ * @param {BidRequest} bidRequest bid request object
+ * @param {BidderRequest} bidderRequest bidder request object
+ * @return Object OpenRTB's 'site' object
  */
-function isSafeFrame() {
-  return window.$sf && window.$sf.ext;
-}
+function openRtbSite(bidRequest, bidderRequest) {
+  let result = {};
 
-/**
- * @returns true if the environment is a dfp safe frame.
- */
-function isDFPSafeFrame() {
-  if (window.location && window.location.href) {
-    const href = window.location.href;
-    return (
-      isSafeFrame() &&
-      href.indexOf('google') !== -1 &&
-      href.indexOf('safeframe') !== -1
-    );
+  const loc = utils.parseUrl(utils.deepAccess(bidderRequest, 'refererInfo.referer'));
+  if (!utils.isEmpty(loc)) {
+    // result.page = `${loc.protocol}://${loc.hostname}${loc.pathname}`;  // commented because otherwise it doesn't work for QA rig page
+  }
+
+  if (self === top && document.referrer) {
+    result.ref = document.referrer;
   }
-  return false;
+
+  const keywords = document.getElementsByTagName('meta')['keywords'];
+  if (keywords && keywords.content) {
+    result.keywords = keywords.content;
+  }
+
+  const siteParams = utils.deepAccess(bidRequest, 'params.site');
+  if (siteParams) {
+    Object.keys(siteParams)
+      .filter(param => includes(OPENRTB_VIDEO_SITEPARAMS, param))
+      .forEach(param => result[param] = siteParams[param]);
+  }
+  return result;
 }
 
 /**
- * Return true if we are in an iframe and can't access the top window.
+ * @return Object OpenRTB's 'device' object
  */
-function isSandboxedIframe() {
-  return window.top !== window && !window.frameElement;
+function openRtbDevice() {
+  return {
+    ua: 'Mozilla/5.0 Android', // navigator.userAgent, // hardcoded because otherwise it doesn't work for QA rig page
```

Comment:
`ua` is hardcoded to allow testing with QA rig page on desktops. Should be reverted for real Prebid.js PR.

Reply:
super weird that we support mozzila on android 5.0 xD

Reply:
:D yeah, it's fun that it worked)